index.js

```diff
@@ -45,8 +46,23 @@ class vehicle {
             }
 
             if (result.status == 200) {
-                this.token = result.data.access_token
-                resolve(result.data.access_token)
+                var tokenOptions = {
+                    method: 'PUT',
+                    baseURL: tokenUrl,
+                    url: '/api/oauth2/v1/token',
+                    headers: Object.fromEntries(fordHeaders),
+                    data: { code: result.data.access_token }
+                }
+                console.log(tokenOptions)
+                try {
+                    var tokenResult = await request(tokenOptions)
+                    console.log(tokenResult.data)
+                } catch (err)  {
+                    console.log(err)
+                    reject(err.result.status)
```

Comment:
Also we must return after reject! Otherwise all the code below will still be executed and we might access attributes of null objects.

```suggestion
                    reject(err.result.status)
                    return
```